#167727753 Pre-screened accommodation supplier can add accommodation

tobslob opened the pull request:
## What does this PR do?

PR that enable only pre-screened accommodation supplier to create accommodation

## Description of Task to be completed?

- create accommodation-supplier role
- update create accommodation controller
- seed accommodation supplier user
- test create accommodation endpoint
- update accommodation route

## How should this be manually tested?
- CLONE this repo
- open the cloned directory
- run ``` npm install ```
- run ``` npm run start:dev ```
- access the docs using {base-url}/docs
- POST accommodation using {base-url}/accommodation

## What are the relevant pivotal tracker stories?

[Finishes #167727753](https://www.pivotaltracker.com/story/show/167727753)

tejiri4 commented on src/controllers/accommodationController.js:
```diff
@@ -17,8 +18,16 @@ const { serverError } = messages;
 */
 const createAccommodation = async (req, res) => {
+  const { TRAVEL_ADMIN, SUPER_ADMIN, ACCOMMODATION_SUPPLIER } = roles;
+  const checkRole = TRAVEL_ADMIN || SUPER_ADMIN || ACCOMMODATION_SUPPLIER;
```
LET TRY USING AUTHORIZE MIDDLEWARE ONLY AT FIRST AND SEE IF IT STILL WORKS

tobslob replied:
It works

tobslob committed:
feat(Add accommodation): Pre-screened accommodation supplier can add accommodation

- create accommodation-supplier role
- update create accommodation controller
- seed accommodation supplier user
- test create accommodation endpoint
- update accommodation route

[Finishes #167727753](https://www.pivotaltracker.com/story/show/167727753)

tejiri4 commented on src/models/accommodation.js:
```diff
@@ -47,7 +47,7 @@ module.exports = (sequelize, DataTypes) => {
     },
     accommodationType: {
-      type: DataTypes.ARRAY(DataTypes.STRING),
+      type: DataTypes.STRING,
```
Why did you changed  the data type?

tobslob replied:
I realized it should be a single type... depending on the type the vendor wants to provide

tejiri4 commented:
Do look into the comment I dropped and Also rebase

tejiri4 commented:
Try your PR on heroku using the review app and add a comment if your endpoint is working on heroku

tejiri4 reviewed (approved):
LGTM
